docs/guide/decompositions.md

```
@@ -0,0 +1,86 @@
+```
+
+For matrices already in upper Hessenberg form, `qr_hessenberg` is faster and is
+what the eigensolver uses internally:
+
+```cpp
+static constexpr auto result = consteig::qr_hessenberg(H);
+```
+
+## Hessenberg Reduction
+
+Reduces a square matrix to upper Hessenberg form via orthogonal similarity: H = P^T * A * P. Hessenberg matrices have zeros below the first subdiagonal.
```

markdownlint-cli2 (0.22.0)

[warning] Spaces inside emphasis markers

(MD037, no-space-in-emphasis)